### PR TITLE
Added patch generic resource operation

### PR DESCRIPTION
**Pull request by jaredmoo**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by jaredmoo** [comment]
Working on the .NET SDK test right now - don't merge until that's available. Thx

**Comment by jaredmoo** [comment]
Semantic validation errors in specification/resources/resource-manager/Microsoft.Resources/2015-11-01/resources.json are not my fault but I'm fixing them to make the build clean

**Comment by jaredmoo** [comment]
.NET SDK tests: https://github.com/jaredmoo/azure-sdk-for-net/tree/patchGenericResource

File: specification/resources/resource-manager/Microsoft.Resources/2015-11-01/resources.json
Comment (olydis) [inline specification/resources/resource-manager/Microsoft.Resources/2015-11-01/resources.json]: in case you have bandwidth, I'd strongly suggest factoring out common parameters like this one into the `parameters` section. Much deduplication! 🙂 In that case, don't forget to add `x-ms-parameter-location: method` so the parameter stays on the method and doesn't become a client property

Comment (jaredmoo) [reply specification/resources/resource-manager/Microsoft.Resources/2015-11-01/resources.json]: Understood - I will leave refactoring to the resources team.

**Review by olydis — APPROVED** [review]
LGTM, thanks a ton for fixing the semantic errors! 👍

**Comment by olydis** [comment]
@jaredmoo did [this](https://github.com/Azure/azure-rest-api-specs/pull/2061#issuecomment-348046501) mean that you're done working on the .NET SDK tests? Let me know when you're good to merge 🙂

**Comment by jaredmoo** [comment]
Yes, the SDK tests are working so it's good to merge from my point of view, but we also need signoff from somebody in ARM / resources team (I am in SQL team).

**Comment by olydis** [comment]
indeed, on it

**Comment by olydis** [comment]
@simongdavies see new operations

**Comment by olydis** [comment]
@ravbhatnagar

**Comment by ravbhatnagar** [comment]
Looks fine.

**Pull request merged**
